indexbuilder.go

```diff
@@ -258,14 +269,25 @@ type Document struct {
 	SkipReason string
 
 	// Document sections for symbols. Offsets should use bytes.
-	Symbols []DocumentSection
+	Symbols         []DocumentSection
+	SymbolsMetaData []*Symbol
```

Comment:
I wonder if DocumentSection should be extended to have the symbol data?

Reply:
Doesn't change on disk representation. We want DocumentSection to be small and together on disk for locality ++ perf.